src/app.js

```diff
@@ -383,6 +384,25 @@ function run () {
   })
   var offsetToLineColumnConverter = new OffsetToLineColumnConverter(compiler.event)
 
+  // ---------------- ContextualListener -----------------------
+  this._components.contextualListener = new ContextualListener({
+    getCursorPosition: () => {
+      return this._components.editor.getCursorPosition()
+    },
+    getCompilationResult: () => {
+      return compiler.lastCompilationResult
+    },
+    warnExpression: (position) => {
```

Comment:
Why is it called "warn"?

Reply:
Means that the module wants to warn app.js that this position need a particular treatment. cause we might extend this function to do something else than just highlight.

Reply:
I think currently, this is really confusing. It sounds like this would be used for compiler warnings.

Reply:
Isn't "highlight" actually more general than "warn"?